app/components/small-cart-toggle-view.js

```diff
@@ -2,10 +2,21 @@ import Ember from "ember";
 
 export default Ember.Component.extend({
   showCartSummary: false,
+  allowClick: false,
+
+  application: Ember.computed(function() {
+    return Ember.getOwner(this).lookup("controller:application");
```

Comment:
I don't think that it's a good idea to bind component with controller. 
You can inject a controller in a controller but ember specifically doesn't allow you to inject a controller in a component. 

Is there any other or better way to do this?

Reply:
It's definitely not, but I'm trying to find the path of least resistance.
We have 4 or 5 different implementations of the cart, with just enough differences to make it hard to use a single component everywhere. The action is in the application, so my thought was either re-write it here, or call the one from the application.

I think both ways are equally bad here :/

Reply:
any ideas ?

Reply:
The main point is that I don't want to have yet another implementation of the list, which is why I'm using  `{{partial "cart_content/items"}}` inside the component

Reply:
Will it be possible to move ** showItemDetails** action from application controller to `cart-service` and inject that service in application controller(might be there already) and component?